pkg/cache/cluster_test.go

```diff
@@ -155,20 +115,33 @@ func getChildren(cluster *clusterCache, un *unstructured.Unstructured) []*Resour
 }
 
 func TestEnsureSynced(t *testing.T) {
-	obj1 := strToUnstructured(`
-  apiVersion: apps/v1
-  kind: Deployment
-  metadata: {"name": "helm-guestbook1", "namespace": "default1"}
-`)
-	obj2 := strToUnstructured(`
-  apiVersion: apps/v1
-  kind: Deployment
-  metadata: {"name": "helm-guestbook2", "namespace": "default2"}
-`)
-
-	cluster := newCluster(obj1, obj2)
+	obj1 := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "helm-guestbook1",
+			Namespace: "default1",
+		},
+	}
+	obj2 := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "helm-guestbook2",
+			Namespace: "default2",
+		},
+	}
+
+	cluster := newCluster(t, obj1, obj2)
 	err := cluster.EnsureSynced()
-	assert.Nil(t, err)
+	require.NoError(t, err)
```

Comment:
It was annoying to debug because of a lot of noise. This ensures we see only the first failure and the test stops here, not generating any noise.